net: update testnet torv3 hardcoded seeds for 24.x

jonatack opened the pull request:
As a follow-up to https://github.com/bitcoin/bitcoin/issues/13550 and #22060, replace the mostly unreachable testnet torv3 hardcoded seeds from v0.22 with new ones that are consistently reachable recently and that have service bit 1 set.

This needs to be done before v24.0 to make sure onion-only testnet nodes can still connect to the network.  

Ways to test:

- Re-generate `src/chainparamsseeds.h` with `cd contrib/seeds && python3 generate-seeds.py . > ../../src/chainparamsseeds.h`, check if git tree stays the same
- Re-compile and create a new testnet node with `bitcoind -testnet -dnsseed=0 -onlynet=onion -proxy=127.0.0.1:9050` (or delete `~/.bitcoin/testnet3/peers.dat` and launch bitcoind with `-testnet -dnsseed=0`). Make sure there are no `addnode=` in your `bitcoin.conf`. The debug log should print "Adding fixed seeds".  Check if the node is able to connect to the network and get blocks with for ex. `watch -t ./src/bitcoin-cli -testnet -rpcwait -netinfo 4`
- Check the addrman contains the seeds by running for ex. `bitcoin-cli -rpcwait -testnet getnodeaddresses 0 onion | jq -r '.[] | (.address + ":" + (.port|tostring) + " " + (.services|tostring))' | sort`

- Check if the addresses are connectable, for ex. with this python script by laanwj:

```python3
#!/usr/bin/env python3
import pprint
import subprocess
with open('contrib/seeds/nodes_test.txt') as f:
    for line in (line for line in (line.rstrip().split('#', 1)[0] for line in f) if line):
        pprint.pprint(line)
        subprocess.call(["nc", "-v", "-x", "127.0.0.1:9050", "-z"] + line.split(':'))
```

Thanks to satsie (Stacie Waleyko) for help with the list.

jonatack committed:
contrib: update testnet torv3 hardcoded seeds

- remove unreachable seeds
- add reachable seeds having service bit 1
- update chainparamsseeds.h

Co-authored-by: "Stacie <staciewaleyko@gmail.com>"

satsie reviewed (approved):
ACK - 4cc2dc4

Below are my test steps. I had one question on proposing more seeds, please see the very end of this comment.

1. Check out this PR branch and run build locally. Run the script in this PR description from the root directory. Output:

```
'kf4qlhek34b3kgyxyodlmvgm4bxfrjsbjtgayyaiuyhr2eoyfgtm3bad.onion:18333'
Connection to kf4qlhek34b3kgyxyodlmvgm4bxfrjsbjtgayyaiuyhr2eoyfgtm3bad.onion 18333 port [tcp/*] succeeded!
'mc7k47ndjvvhcgs54wmjzxvate4rtuybbjoryikdssjhcxlx27psbyqd.onion:18333'
Connection to mc7k47ndjvvhcgs54wmjzxvate4rtuybbjoryikdssjhcxlx27psbyqd.onion 18333 port [tcp/*] succeeded!
'z3j5foswuhpmtrg3kb56stkzmuoaesvd5jz3eztq46c4cidapglcyuad.onion:18333'
Connection to z3j5foswuhpmtrg3kb56stkzmuoaesvd5jz3eztq46c4cidapglcyuad.onion 18333 port [tcp/*] succeeded!
```

2.Start up a Tor proxy using the Tor browser and launch `bitcoind` in testnet mode behind the Tor proxy:

```
./src/bitcoind -testnet -proxy=127.0.0.1:9050
```

3. Manually add the nodes:

```
./src/bitcoin-cli -testnet addnode "z3j5foswuhpmtrg3kb56stkzmuoaesvd5jz3eztq46c4cidapglcyuad.onion:18333" "add"
./src/bitcoin-cli -testnet addnode "mc7k47ndjvvhcgs54wmjzxvate4rtuybbjoryikdssjhcxlx27psbyqd.onion:18333" "add"
./src/bitcoin-cli -testnet addnode "kf4qlhek34b3kgyxyodlmvgm4bxfrjsbjtgayyaiuyhr2eoyfgtm3bad.onion:18333" "add"
```

4. From the `bitcoind` logs:

```
2022-08-16T02:02:58Z New outbound peer connected: version: 70016, blocks=2340859, peer=4 (manual)
2022-08-16T02:04:00Z New outbound peer connected: version: 70016, blocks=2340860, peer=8 (manual)
2022-08-16T02:04:14Z New outbound peer connected: version: 70016, blocks=2340860, peer=9 (manual)
```

5. Double check the nodes were added:

```
./src/bitcoin-cli -testnet getpeerinfo | grep 'onion'
    "addr": "z3j5foswuhpmtrg3kb56stkzmuoaesvd5jz3eztq46c4cidapglcyuad.onion:18333",
    "network": "onion",
    "addr": "mc7k47ndjvvhcgs54wmjzxvate4rtuybbjoryikdssjhcxlx27psbyqd.onion:18333",
    "network": "onion",
    "addr": "kf4qlhek34b3kgyxyodlmvgm4bxfrjsbjtgayyaiuyhr2eoyfgtm3bad.onion:18333",
    "network": "onion",
```

I checked to see if my testnet node knew anymore and was surprised to see 192 results:

```
./src/bitcoin-cli -testnet getnodeaddresses 0 onion | jq length
192
```

Is there a good way to evaluate which are stable? Should I add some (how many is reasonable?) to nodes_test.txt and run that script you wrote (in this PR description) for a few days? If they are still up, move ahead with proposing them (either with a comment in this PR or a new one?)

jonatack commented:
Nice work!

> Is there a good way to evaluate which are stable? Should I add some (how many is reasonable?) to nodes_test.txt and run that script you wrote (in this PR description) for a few days? If they are still up, move ahead with proposing them (either with a comment in this PR or a new one?)

Yes, we could use more and I'm looking at adding some here (we can compare notes :). As for criteria, have a look at the bottom of `contrib/seeds/makeseeds.py`; it suggests filtering by blocks received, service bit, uptime, known and recent user agent, etc.  For `nodes_main_manual.txt` I mostly proposed peers run by trusted colleagues that I could ask, if needed, whether they intended to maintain the node over time. I have only been keeping track of that kind of info for mainnet, but mentioning it in case you know some of your testnet peers.

jonatack commented:
(I forgot to update `src/chainparamsseeds.h` like in https://github.com/bitcoin/bitcoin/pull/22060; will update when we add more seeds here.)

satsie commented:
Thanks for that additional info! Unfortunately I don't know any of my testnet peers so I tried what I could to weed unreliable peers out of the list I got from my node yesterday.

If I'm not mistaken, the statistics on blocks received, service bit, uptime, known and recent user agent, etc. (the ones that `makeseeds.py` looks at) all come from `seeds_main.txt` which, according to the README, comes from https://bitcoin.sipa.be/seeds.txt.gz

I take it the only way to gather this info is by running the crawler (https://github.com/sipa/bitcoin-seeder)? That stuff is over my head but I did a few a sanity checks on the 190+ onion addresses I gathered last night.

I added the list addresses to `nodes_test.txt` with the help of this command:

```
./src/bitcoin-cli -testnet getnodeaddresses 0 onion | jq -r '.[] | (.address + ":" + (.port|tostring))'
```

Then I ran the script that is in this PR description and removed the (76) unreachable ones from the list. This brought the count down from 195 to 119. A few hours later I ran the script one more time to make sure the peers were still up. 7 of them were unreachable which brought my list down to 112 addresses. After removing some duplicates, I was down to 106.

<details>
  <summary>Here is a list of the ones I was left with:</summary>
  
```  
hqgoy62hoqjmz37brdfvoeov3cix5fixbqjoert4ydr6herg5oc3iwyd.onion:18333 
pm57didyzg5ljuvn5ufr5uun2iencuk3af2gzqc5zvgfh452c3rxtjyd.onion:18333 
zmvizz7fd5hdue6wt3lwqumd6qwt4ijymmmotfzh75curq3mzjm53hyd.onion:18333 
zcep44k7unwjm2wxty4ijh2e4fv5zgbrvwlctzyaqnrqhltjfzrtodad.onion:18333 
bdaipyhbcpyztzcujx3fwlsgopobr6l6wb4ct2zdklbdbh3lo3i4hmad.onion:8333 
wzpdt24tdark26eugredddorik3tqwcj5ialtt2yim4ceiuiq7phkyqd.onion:18333 
3pe3fyklipy4sppkkgnhc22kcxtt57uler5kv72t676bbrwmcseo5qad.onion:18333 
33qtvikhnsziv5geusi7dem4q6upi2njqm7ljq57fyh64vbpffeh7yid.onion:18333 
xgapnikkbldoggjh5ewxkyauhuwnvf3xkspxroe3ojvfrk4lswkyx5yd.onion:18333 
gy6nih4pmp5esyvvnhlj6qvk7zkbjuoswkxffyiip3dbkvsfxwz5zcqd.onion:18333 
67s3af64ehw7xnxv422axm7tns4d6kutrftc6bjq375n74q3kj4pp7ad.onion:18333 
7v2ja4igx4v5y2jr6jrr6gaxohjhlzhvgwe4avlraxchozf7ea3kruqd.onion:18333 
oaiw2lnhzgp5ry7ivzneuufmh7lfploquu2rjv5rozmlbefedsnxe5qd.onion:18333 
bluk62wj24bsvdwh47muo54hhwsatkftiqxevt5kba7hstjoex6ueeyd.onion:18333 
aprzvj7hgctsde4mkj3ewq35gvykspjvkqiygg7bpnw5tkvse2n7rhid.onion:18333 
hbkp5xwpqo4qm75kpglfrclyiuuvdgv7mtiqfys7oqks4dmpqgpeoeid.onion:18333 
xkvzdhcirontixbq6pjhru57bf4sgtqylvphk25csfrsy5p5ay3oc3yd.onion:18333 
rxilsibapu7f5emeqjr4rlmnxdqqx6mu7spqyqk7hwiim6mmeyshruyd.onion:18333 
otmfnhc6wrrbf2tpdy6zkisqc3r3urnsuowsnmatoto6yixaocnkseid.onion:18333 
7v6uybtzvpcfelsneso3w26omgqiwttfqh7dd5mn4ihuz74bnwib2jad.onion:18333 
ujdchuw3hz5gkbouiv4p6pwbfdn7v4k6gluwvd4wiukqc7y7ow754uad.onion:18333 
yo5fd6atmhqh6wrytm6xttxtbu5u52avzbmzidagtxyyz3rv44ey4zad.onion:18333 
qe2jbe447he6panfvpyqhyntf7346gmuf55bxrmdzggmgwyjsyknhxyd.onion:18333 
xnipauenw5wnjb2zbx6v6umgvbb3g6xhf5kjo7pnyn5tdzvzaxtzicid.onion:18333 
ayynqazucyh2jd5rehcfggmhunqpdwzlbhzbqgy6lj4ctz2ocj7chpid.onion:18333 
o65fsury4rs44qpq3kzlz5z6fupngysptis53grqnch26a6xvszaijid.onion:18333 
24j74ahq6ed4wmfrghdwroyfzimlkhnrb7zh4zw3vl2allzxbjrhaqid.onion:18333 
reswkos4eh6x4tw3u2p37bwowdcfsvmt6sazspaljc5odzm6b7s53cad.onion:18333 
ew7havgwsrjdu7xyiloy2tvtdg5cqjj4x62ojtem2nt2w7l4jadq7bad.onion:18333 
gsw6sn27quwf6u3swgra6o7lrp5qau6kt3ymuyoxgkth6wntzm2bjwyd.onion:18333 
adstabjz7ec2y3jt4w2dvummowzv7g6m2f3kajeejffuaz7ojwj6epqd.onion:18333 
zcsn3j6aswnrf56xj2n5jn4tlwmyoq4benjn5ujcgz46co2y5tm642id.onion:18333 
f47iwce2koob3wsq2buwd24wfv6zar3vezvoa4yi32ugnd3xc33p77qd.onion:18333 
imya36iexiiiqrkwuxxcehnv4kg5shtirwd2vg4cnjy6lfjlph3fusqd.onion:18333 
bubm6fiopfzkxqrfx6vqpioe5ahlhyubz57ogsqqy4ha5pnngiqlh6id.onion:18333 
ddj4cuvb32ve5chtp6jattcdnnmxmpoofjthzi7thgxxht7yqoetj3yd.onion:18333 
zoaa3x7quyuijggii5zl4uyeioodudsgtr2uyv2qtdsslac5ukiwlxid.onion:18333 
rp6pn3b3oesyr2giolbysbjhqeugxntsu7crnkth4y33ok4zvcl7yrqd.onion:18333 
jsc4frvvnl2d3bhzyofsc72xpztgm23nl4fnb4dwkzsxr6fhij2q5iyd.onion:18333 
hvbmmzvqrpgps2x5u4ip4ksf3e5m2fneac754gtnhjn2rsevni6cz3ad.onion:18333 
6a4ony53julvnufo632ktgmwvhupz63wbdwx7n7qudjy32qyq6gm3bqd.onion:18333 
dd5w7bor5ovhz6xyti3dg4ifkeza6j7hzd4sw5ebt7ocrsgxij423vyd.onion:18333 
jjfuyj7krgzkmpxvn3b2j2hwlzkmze3ezy3ifwk7dnswwawgmzqhjrqd.onion:18333 
vctlwaqgmu53eutz2hewuakcipfgtyljsd7czut4dd62xr3rp6fqezad.onion:18333 
ji5wmshokuc63eiulzlwj2zdvnligvrwfvvc76bice3tu43wfzvpmkyd.onion:18333 
fzbrwmgwmko7quelrhfuskt3ijabac76zx7g52dfrevmhdkj6ivh7qyd.onion:18333 
hacjjgj2mbqqrthzimmi6anvin7dljjhfl3ik6ebg3w3nmgsvr3ymmqd.onion:18333 
pmismhpwug34gnqzbutranvx2wjwbshyqj4un2dyzyuvak2eh55psfyd.onion:18333 
oln7ybci53wk4g5n42nipyixvyjxbludsbrfsmhnirb6tk7ovlikd5id.onion:18333 
dwb47cmqa2tjpmvjaear7gdcars2lez6niefhi4qf22qehtyta6577qd.onion:18333 
wnxgjgjgplv5iu4mssyuunycvku4qnqr5t4q6cfdt47k7uwrfifuirad.onion:18333 
nkyqozv6kdwi423s7s2mezzguf5bafot2a3hv4ed2dbvtblisdmad4qd.onion:18333 
i5oftdk6lxq4j65vxfstiqqaqw4znqlcci56xnyodbeusay3b5emxgid.onion:18333 
yejwatt6zi36dnccwfokyihax5bvjolbjgfhj2zkxuguxyruwzssoyad.onion:18333 
iuhhuocns7entrzlxsxktyz2ibs7hqgiggv6sauzqkzka6laslwz7oqd.onion:18333 
iddr66ewkhenivapgianudjkwqcp6dxtssg7ixrdot5az6uh7m5tmjqd.onion:18333 
mjbg3ggeuelmc7ixty3zjccyo2urg2uyherfqe7ytkm2ejkwlec7h6ad.onion:18333 
3nujuvgaauh6srfzuezkdubisavjwwihwi4snkjar4aiusdbirvmrkid.onion:18333 
yda7kwpii33j2qpq32ftf6lp22znknswipjwaccvsqj7l337jvfesnid.onion:18333 
kwjxlauwjtecjfsiwopbl5pvn5n6z5rz76uk6osmlurd3uyuymcw7aid.onion:18333 
2fy74te65gm3c3gv3u5mhwdudvbdfh6k5fdz4gduimrltjjrxftbxrqd.onion:18333 
dmbuz5xohpuiojomtpmapsyp4lw3atznrh6wahdvhvlmwgoxdnz2h3id.onion:18333 
shph2uqcvbe2hc25q3ln73qsche5ddqb4u5tkmgqv73aicrmmkkj4kid.onion:18333 
hmjm7l34olcvkgcvpjkbmpak6445tb7w3uc3ri7ea2uybbxtdzsq37id.onion:18333 
4ljeyu6hptzcqocntjmg2hdvpkc6e2mrcln4e72u45gzuirfdertwdqd.onion:18333 
aesy6tfufadkut6flu2bsqgnw2422ur2ynjalguxlzuzuktg3zehttqd.onion:18333 
2lsncqdflwk272dhydrxf7ikfy23ppnmm54dnynyxiym6lqf3wowrmqd.onion:18333 
etuymy47s3quepvdaoo72i5e5mc7uovrzu5m4jf5q6mwlwizoxy4xgid.onion:18333 
wpkbkdr7clw7zk3jkwiult6bf422j54u77ml4rgig2xq7icogyrcspid.onion:18333 
polarisultijjhaku6z6u7jyboho5epdsg44ttebfaxmgau2z5sqolad.onion:18333 
lc7upz2srw2yhpcvwg4afy64ylcoo6mfwlttqj5ovuglqnhnohpi5iqd.onion:18333 
7554uw5djruh34j5ddx3iprzgqgzypcjtptwoldymfbgoywqcw2wiwyd.onion:18333 
y2cpty5imvnbnkkpjfsxfp6l72jvosskz2vecouixnfq227p2ivpdwyd.onion:18333 
dbosgoqgk5oglsqqboy45cyd65wnszaj7d2lqo5rqllpia4xw7fodnid.onion:18333 
yzojfeq74ocriamp5f7wtpc2erpkwuienrnt4ja2dggl5b52pt7dhxqd.onion:18333 
klymxdvje7kccv3tznabo3udopsftkmjemkbi2urqxjm4hefaudejjyd.onion:18333 
5zlrxk6q24t4vz5k4ie7gtuasdjavhoelhinzimxbfhc77u7vafipsid.onion:18333 
o6vfovqxz3oxszfppczpjejwouobztjrgvfojc3emvhan3bkyskzhuad.onion:18333 
4u4mcz2sfvxs7pwcwncswgmmcdzqtzjx7ztfo332jv4pqucb22ikdhad.onion:18333 
tyo6jvqjh34r2b7i477xlcrwt7rq2fwuwpax5s3nw3tfbtwbox7aakyd.onion:18333 
2hg2r3xraxs2cowyzydvtvfhmr7rknnr2zwe2ijkjba6f7fujnmauoyd.onion:18333 
7blv5abnytdf47yvbhxmykprmvjryqob65i2jmdwq3rrajcn2iiysbqd.onion:18333 
aoeart34umoonvd2kbqr3bc4sweu6a4msh2gp4skyqvei3shzcxbgmyd.onion:18333 
vf5ur53tzmdtotvkndcgochklnuav7quqjvkc6mctqfvef6wnmn26mid.onion:18333 
fbimesnyhzubbzqc3uaufzkbyfmnkxvypoxaveaub7rzpzh2foxrn2yd.onion:18333 
awpk6z3xghx6ozouhodcydaqtr6uzzbnw4creuix7mkupxoxlmhhspad.onion:18333 
be7zx3hh6dlahorlvsrrgqm4oahfrgqm2tbwnbd4u53ntu5f765n6hyd.onion:18333 
7zgbmtzxow2oevd5aaqtsormw7ujv4zprl3oi2355immhq4gk7cyw5ad.onion:18333 
b2ika53aqckv4gs7wmog3byrea2vfzm5p7ye33digcsmvvnpbyqmzoyd.onion:18333 
ap4igrd45iglxbzkr3pzrtmepbui6ul6bzhh5mg74miktj7vmwvy3jid.onion:18333 
lf3mpxfyjuovcqdvinl52pvdmmda6xqyfeiarlfamdjpgy3ouzmmlbyd.onion:18333 
766lozlabxaqjpbqsvt6sn3c65n6gkwwhoxyvggj7nfwnmw4cpaoccad.onion:18333 
y4eud6iabao4666vcq3qch6kvg6lg5q5hazzetk4jnwr6vcdyjdv2yyd.onion:18333 
d3czabzjj57lgrsr5gawkjd7v3gznrqa7zyizqmk4lryascavmipnyad.onion:18333 
e7tkrf54ng3q5vcn5gn77zwjwm74lkfav4mwdux3pvon6yvqg3tf46qd.onion:18333 
zc2lxsl4wp56xkryf37bvqnex25kpszxghhndgc3rm7xmhy3gpzlhtad.onion:18333 
mjuc33cf27xlrafcbuiiscclfb7fxlbi6myrdbzkvrckf2b4w56cjpid.onion:18333 
drthcyb4x4rdfekw5g7xjogxi7aqoluilgulbgwvsme3nw3oibvchbad.onion:18333 
3oo6bsc5mvf6a6ypmoaikilta6ka7mbdhdwhrnqhuhjlbaxyedvfvaqd.onion:18333 
5v3i2kfqiqwp75gznjoptss7qgrcgseceqxpzpqkd34qeqzrg726i7id.onion:18333 
6ftyg3nhc6tn2hyzls6zfdsfbroczhkxtdqumqb5q4yafhy5rdpapbid.onion:18333 
lzga6ljccvnruiocc3p3ky2nu4tuss4zszqvxeuvc26xs7f43kkwjaad.onion:18333 
zovauxlorl5eswumbsoxv2m5y3sm3qlk7657dcpr2uld7xf35en46sqd.onion:18333 
jn2p4sgfphkxpow7kjrubrbqat77kkibzqkvuwhxyalcrazwmcqeaqyd.onion:18333 
dqhhlssfwmh3g6zhwxpcfbw64xz5rfikcglinbhoxv5ajv4qzicjyeid.onion:18333 
hw3vzp32w4h6giplue6ix445oi6wt7gmeksrznb7tdfwhkgit7gnbbad.onion:18333
```
</details>

Let me know if there's anything else I can do to help identify reliable seeds.

Sjors commented on contrib/seeds/nodes_test.txt:
```diff
@@ -1,16 +1,6 @@
-# Onion nodes
-35k2va6vyw4oo5ly2quvcszgdqr56kcnfgcqpnpcffut4jn3mhhwgbid.onion:18333
-blo2esfvk2rr7sr4jspmu3vt2vpgr5rigflsj645fnku7v4qmljurtid.onion:18333
-fuckcswupr5rmlvx2kqqrrosxvjyong4hatmuvxsvtcwe4dsh5rus7qd.onion:18333
```
Such a noble vanity key though :-)

jonatack commented:
Thanks! I tested your list.

<details><summary>These were repeatedly unreachable</summary><p>

```
bdaipyhbcpyztzcujx3fwlsgopobr6l6wb4ct2zdklbdbh3lo3i4hmad.onion:18333
rxilsibapu7f5emeqjr4rlmnxdqqx6mu7spqyqk7hwiim6mmeyshruyd.onion:18333
ew7havgwsrjdu7xyiloy2tvtdg5cqjj4x62ojtem2nt2w7l4jadq7bad.onion:18333
kwjxlauwjtecjfsiwopbl5pvn5n6z5rz76uk6osmlurd3uyuymcw7aid.onion:18333
shph2uqcvbe2hc25q3ln73qsche5ddqb4u5tkmgqv73aicrmmkkj4kid.onion:18333
lc7upz2srw2yhpcvwg4afy64ylcoo6mfwlttqj5ovuglqnhnohpi5iqd.onion:18333
be7zx3hh6dlahorlvsrrgqm4oahfrgqm2tbwnbd4u53ntu5f765n6hyd.onion:18333
```
</p></details>

<details><summary>We both have these reachable peers in common</summary><p>

```
5v3i2kfqiqwp75gznjoptss7qgrcgseceqxpzpqkd34qeqzrg726i7id.onion:18333
5zlrxk6q24t4vz5k4ie7gtuasdjavhoelhinzimxbfhc77u7vafipsid.onion:18333
766lozlabxaqjpbqsvt6sn3c65n6gkwwhoxyvggj7nfwnmw4cpaoccad.onion:18333
adstabjz7ec2y3jt4w2dvummowzv7g6m2f3kajeejffuaz7ojwj6epqd.onion:18333
ayynqazucyh2jd5rehcfggmhunqpdwzlbhzbqgy6lj4ctz2ocj7chpid.onion:18333
bluk62wj24bsvdwh47muo54hhwsatkftiqxevt5kba7hstjoex6ueeyd.onion:18333
drthcyb4x4rdfekw5g7xjogxi7aqoluilgulbgwvsme3nw3oibvchbad.onion:18333
gy6nih4pmp5esyvvnhlj6qvk7zkbjuoswkxffyiip3dbkvsfxwz5zcqd.onion:18333
mjbg3ggeuelmc7ixty3zjccyo2urg2uyherfqe7ytkm2ejkwlec7h6ad.onion:18333
oln7ybci53wk4g5n42nipyixvyjxbludsbrfsmhnirb6tk7ovlikd5id.onion:18333
rp6pn3b3oesyr2giolbysbjhqeugxntsu7crnkth4y33ok4zvcl7yrqd.onion:18333
ujdchuw3hz5gkbouiv4p6pwbfdn7v4k6gluwvd4wiukqc7y7ow754uad.onion:18333
yda7kwpii33j2qpq32ftf6lp22znknswipjwaccvsqj7l337jvfesnid.onion:18333
zcep44k7unwjm2wxty4ijh2e4fv5zgbrvwlctzyaqnrqhltjfzrtodad.onion:18333
```
</p></details>

<details><summary>A combined list of reachable peers I'll test connecting to and check services</summary><p>

```
5v3i2kfqiqwp75gznjoptss7qgrcgseceqxpzpqkd34qeqzrg726i7id.onion:18333
5zlrxk6q24t4vz5k4ie7gtuasdjavhoelhinzimxbfhc77u7vafipsid.onion:18333
766lozlabxaqjpbqsvt6sn3c65n6gkwwhoxyvggj7nfwnmw4cpaoccad.onion:18333
adstabjz7ec2y3jt4w2dvummowzv7g6m2f3kajeejffuaz7ojwj6epqd.onion:18333
ayynqazucyh2jd5rehcfggmhunqpdwzlbhzbqgy6lj4ctz2ocj7chpid.onion:18333
bluk62wj24bsvdwh47muo54hhwsatkftiqxevt5kba7hstjoex6ueeyd.onion:18333
drthcyb4x4rdfekw5g7xjogxi7aqoluilgulbgwvsme3nw3oibvchbad.onion:18333
gy6nih4pmp5esyvvnhlj6qvk7zkbjuoswkxffyiip3dbkvsfxwz5zcqd.onion:18333
kf4qlhek34b3kgyxyodlmvgm4bxfrjsbjtgayyaiuyhr2eoyfgtm3bad.onion:18333
mc7k47ndjvvhcgs54wmjzxvate4rtuybbjoryikdssjhcxlx27psbyqd.onion:18333
mjbg3ggeuelmc7ixty3zjccyo2urg2uyherfqe7ytkm2ejkwlec7h6ad.onion:18333
oln7ybci53wk4g5n42nipyixvyjxbludsbrfsmhnirb6tk7ovlikd5id.onion:18333
rp6pn3b3oesyr2giolbysbjhqeugxntsu7crnkth4y33ok4zvcl7yrqd.onion:18333
ujdchuw3hz5gkbouiv4p6pwbfdn7v4k6gluwvd4wiukqc7y7ow754uad.onion:18333
yda7kwpii33j2qpq32ftf6lp22znknswipjwaccvsqj7l337jvfesnid.onion:18333
z3j5foswuhpmtrg3kb56stkzmuoaesvd5jz3eztq46c4cidapglcyuad.onion:18333
zcep44k7unwjm2wxty4ijh2e4fv5zgbrvwlctzyaqnrqhltjfzrtodad.onion:18333
kwjxlauwjtecjfsiwopbl5pvn5n6z5rz76uk6osmlurd3uyuymcw7aid.onion:18333
alxo32b5edi3bn2e224qrgytgxxpic4knyipvpdvctfsrvcaiq5lgeyd.onion:18333
33qtvikhnsziv5geusi7dem4q6upi2njqm7ljq57fyh64vbpffeh7yid.onion:18333
33o6qaidta7s2pmltet6vynd337vamgcifhh44rehwwxqpflcjt2njid.onion:18333
hqgoy62hoqjmz37brdfvoeov3cix5fixbqjoert4ydr6herg5oc3iwyd.onion:18333
pm57didyzg5ljuvn5ufr5uun2iencuk3af2gzqc5zvgfh452c3rxtjyd.onion:18333
zmvizz7fd5hdue6wt3lwqumd6qwt4ijymmmotfzh75curq3mzjm53hyd.onion:18333
wzpdt24tdark26eugredddorik3tqwcj5ialtt2yim4ceiuiq7phkyqd.onion:18333
3pe3fyklipy4sppkkgnhc22kcxtt57uler5kv72t676bbrwmcseo5qad.onion:18333
xgapnikkbldoggjh5ewxkyauhuwnvf3xkspxroe3ojvfrk4lswkyx5yd.onion:18333
67s3af64ehw7xnxv422axm7tns4d6kutrftc6bjq375n74q3kj4pp7ad.onion:18333
7v2ja4igx4v5y2jr6jrr6gaxohjhlzhvgwe4avlraxchozf7ea3kruqd.onion:18333
oaiw2lnhzgp5ry7ivzneuufmh7lfploquu2rjv5rozmlbefedsnxe5qd.onion:18333
aprzvj7hgctsde4mkj3ewq35gvykspjvkqiygg7bpnw5tkvse2n7rhid.onion:18333
hbkp5xwpqo4qm75kpglfrclyiuuvdgv7mtiqfys7oqks4dmpqgpeoeid.onion:18333
xkvzdhcirontixbq6pjhru57bf4sgtqylvphk25csfrsy5p5ay3oc3yd.onion:18333
otmfnhc6wrrbf2tpdy6zkisqc3r3urnsuowsnmatoto6yixaocnkseid.onion:18333
7v6uybtzvpcfelsneso3w26omgqiwttfqh7dd5mn4ihuz74bnwib2jad.onion:18333
yo5fd6atmhqh6wrytm6xttxtbu5u52avzbmzidagtxyyz3rv44ey4zad.onion:18333
qe2jbe447he6panfvpyqhyntf7346gmuf55bxrmdzggmgwyjsyknhxyd.onion:18333
xnipauenw5wnjb2zbx6v6umgvbb3g6xhf5kjo7pnyn5tdzvzaxtzicid.onion:18333
o65fsury4rs44qpq3kzlz5z6fupngysptis53grqnch26a6xvszaijid.onion:18333
24j74ahq6ed4wmfrghdwroyfzimlkhnrb7zh4zw3vl2allzxbjrhaqid.onion:18333
reswkos4eh6x4tw3u2p37bwowdcfsvmt6sazspaljc5odzm6b7s53cad.onion:18333
gsw6sn27quwf6u3swgra6o7lrp5qau6kt3ymuyoxgkth6wntzm2bjwyd.onion:18333
zcsn3j6aswnrf56xj2n5jn4tlwmyoq4benjn5ujcgz46co2y5tm642id.onion:18333
f47iwce2koob3wsq2buwd24wfv6zar3vezvoa4yi32ugnd3xc33p77qd.onion:18333
imya36iexiiiqrkwuxxcehnv4kg5shtirwd2vg4cnjy6lfjlph3fusqd.onion:18333
bubm6fiopfzkxqrfx6vqpioe5ahlhyubz57ogsqqy4ha5pnngiqlh6id.onion:18333
ddj4cuvb32ve5chtp6jattcdnnmxmpoofjthzi7thgxxht7yqoetj3yd.onion:18333
zoaa3x7quyuijggii5zl4uyeioodudsgtr2uyv2qtdsslac5ukiwlxid.onion:18333
jsc4frvvnl2d3bhzyofsc72xpztgm23nl4fnb4dwkzsxr6fhij2q5iyd.onion:18333
hvbmmzvqrpgps2x5u4ip4ksf3e5m2fneac754gtnhjn2rsevni6cz3ad.onion:18333
6a4ony53julvnufo632ktgmwvhupz63wbdwx7n7qudjy32qyq6gm3bqd.onion:18333
dd5w7bor5ovhz6xyti3dg4ifkeza6j7hzd4sw5ebt7ocrsgxij423vyd.onion:18333
jjfuyj7krgzkmpxvn3b2j2hwlzkmze3ezy3ifwk7dnswwawgmzqhjrqd.onion:18333
vctlwaqgmu53eutz2hewuakcipfgtyljsd7czut4dd62xr3rp6fqezad.onion:18333
ji5wmshokuc63eiulzlwj2zdvnligvrwfvvc76bice3tu43wfzvpmkyd.onion:18333
fzbrwmgwmko7quelrhfuskt3ijabac76zx7g52dfrevmhdkj6ivh7qyd.onion:18333
hacjjgj2mbqqrthzimmi6anvin7dljjhfl3ik6ebg3w3nmgsvr3ymmqd.onion:18333
pmismhpwug34gnqzbutranvx2wjwbshyqj4un2dyzyuvak2eh55psfyd.onion:18333
dwb47cmqa2tjpmvjaear7gdcars2lez6niefhi4qf22qehtyta6577qd.onion:18333
wnxgjgjgplv5iu4mssyuunycvku4qnqr5t4q6cfdt47k7uwrfifuirad.onion:18333
nkyqozv6kdwi423s7s2mezzguf5bafot2a3hv4ed2dbvtblisdmad4qd.onion:18333
yejwatt6zi36dnccwfokyihax5bvjolbjgfhj2zkxuguxyruwzssoyad.onion:18333
iuhhuocns7entrzlxsxktyz2ibs7hqgiggv6sauzqkzka6laslwz7oqd.onion:18333
iddr66ewkhenivapgianudjkwqcp6dxtssg7ixrdot5az6uh7m5tmjqd.onion:18333
3nujuvgaauh6srfzuezkdubisavjwwihwi4snkjar4aiusdbirvmrkid.onion:18333
2fy74te65gm3c3gv3u5mhwdudvbdfh6k5fdz4gduimrltjjrxftbxrqd.onion:18333
dmbuz5xohpuiojomtpmapsyp4lw3atznrh6wahdvhvlmwgoxdnz2h3id.onion:18333
4ljeyu6hptzcqocntjmg2hdvpkc6e2mrcln4e72u45gzuirfdertwdqd.onion:18333
aesy6tfufadkut6flu2bsqgnw2422ur2ynjalguxlzuzuktg3zehttqd.onion:18333
etuymy47s3quepvdaoo72i5e5mc7uovrzu5m4jf5q6mwlwizoxy4xgid.onion:18333
wpkbkdr7clw7zk3jkwiult6bf422j54u77ml4rgig2xq7icogyrcspid.onion:18333
polarisultijjhaku6z6u7jyboho5epdsg44ttebfaxmgau2z5sqolad.onion:18333
7554uw5djruh34j5ddx3iprzgqgzypcjtptwoldymfbgoywqcw2wiwyd.onion:18333
y2cpty5imvnbnkkpjfsxfp6l72jvosskz2vecouixnfq227p2ivpdwyd.onion:18333
dbosgoqgk5oglsqqboy45cyd65wnszaj7d2lqo5rqllpia4xw7fodnid.onion:18333
yzojfeq74ocriamp5f7wtpc2erpkwuienrnt4ja2dggl5b52pt7dhxqd.onion:18333
klymxdvje7kccv3tznabo3udopsftkmjemkbi2urqxjm4hefaudejjyd.onion:18333
o6vfovqxz3oxszfppczpjejwouobztjrgvfojc3emvhan3bkyskzhuad.onion:18333
4u4mcz2sfvxs7pwcwncswgmmcdzqtzjx7ztfo332jv4pqucb22ikdhad.onion:18333
tyo6jvqjh34r2b7i477xlcrwt7rq2fwuwpax5s3nw3tfbtwbox7aakyd.onion:18333
2hg2r3xraxs2cowyzydvtvfhmr7rknnr2zwe2ijkjba6f7fujnmauoyd.onion:18333
7blv5abnytdf47yvbhxmykprmvjryqob65i2jmdwq3rrajcn2iiysbqd.onion:18333
aoeart34umoonvd2kbqr3bc4sweu6a4msh2gp4skyqvei3shzcxbgmyd.onion:18333
vf5ur53tzmdtotvkndcgochklnuav7quqjvkc6mctqfvef6wnmn26mid.onion:18333
fbimesnyhzubbzqc3uaufzkbyfmnkxvypoxaveaub7rzpzh2foxrn2yd.onion:18333
awpk6z3xghx6ozouhodcydaqtr6uzzbnw4creuix7mkupxoxlmhhspad.onion:18333
7zgbmtzxow2oevd5aaqtsormw7ujv4zprl3oi2355immhq4gk7cyw5ad.onion:18333
b2ika53aqckv4gs7wmog3byrea2vfzm5p7ye33digcsmvvnpbyqmzoyd.onion:18333
ap4igrd45iglxbzkr3pzrtmepbui6ul6bzhh5mg74miktj7vmwvy3jid.onion:18333
lf3mpxfyjuovcqdvinl52pvdmmda6xqyfeiarlfamdjpgy3ouzmmlbyd.onion:18333
y4eud6iabao4666vcq3qch6kvg6lg5q5hazzetk4jnwr6vcdyjdv2yyd.onion:18333
d3czabzjj57lgrsr5gawkjd7v3gznrqa7zyizqmk4lryascavmipnyad.onion:18333
e7tkrf54ng3q5vcn5gn77zwjwm74lkfav4mwdux3pvon6yvqg3tf46qd.onion:18333
zc2lxsl4wp56xkryf37bvqnex25kpszxghhndgc3rm7xmhy3gpzlhtad.onion:18333
mjuc33cf27xlrafcbuiiscclfb7fxlbi6myrdbzkvrckf2b4w56cjpid.onion:18333
3oo6bsc5mvf6a6ypmoaikilta6ka7mbdhdwhrnqhuhjlbaxyedvfvaqd.onion:18333
6ftyg3nhc6tn2hyzls6zfdsfbroczhkxtdqumqb5q4yafhy5rdpapbid.onion:18333
lzga6ljccvnruiocc3p3ky2nu4tuss4zszqvxeuvc26xs7f43kkwjaad.onion:18333
zovauxlorl5eswumbsoxv2m5y3sm3qlk7657dcpr2uld7xf35en46sqd.onion:18333
jn2p4sgfphkxpow7kjrubrbqat77kkibzqkvuwhxyalcrazwmcqeaqyd.onion:18333
dqhhlssfwmh3g6zhwxpcfbw64xz5rfikcglinbhoxv5ajv4qzicjyeid.onion:18333
hw3vzp32w4h6giplue6ix445oi6wt7gmeksrznb7tdfwhkgit7gnbbad.onion:18333
```
</p></details>

theStack commented:
Concept ACK

satsie commented:
Great! I just ran the script for the combined list and was able to connect to everybody except these three: 

```
3nujuvgaauh6srfzuezkdubisavjwwihwi4snkjar4aiusdbirvmrkid.onion:18333
4ljeyu6hptzcqocntjmg2hdvpkc6e2mrcln4e72u45gzuirfdertwdqd.onion:18333
2hg2r3xraxs2cowyzydvtvfhmr7rknnr2zwe2ijkjba6f7fujnmauoyd.onion:18333
```

jonatack commented:
Thanks @satsie for your help. Updated after several more days of checking uptime, filtered the remaining seeds for service bit 1 and updated `src/chainparamsseeds.h` and the pull description.

satsie commented:
Woo hoo! Thanks for the updated list and test instructions. Can you explain service bit part? I see it in `seeds_main.txt` (and `makeseeds.py` reads the service bit), but how did you get it for the tor nodes? Is it from the "services" field in the `getnodeaddresses` rpc?

It's probably best to have another, different set of eyes on this, but for the record I have tested your changes (`4391364`):

1. Regenerated `chainparamsseeds.h`. No differences between my version and the one in this PR.

2. Re-compiled (by running `autogen.sh`, `configure`, and `make`. I forgot to do this the first time I tried to retest and saw a bunch of failed connections to the unreachable seeds that this PR removes) and brought up a new testnet node with the command you provided (also deleted `testnet3/peers.dat` before launching for good measure). 15 min into startup the logs look good. After a little while I see some failed connections, but a spot check revealed none of these peers were in `nodes_test.txt`.
<details>
  <summary>logs from startup</summary>
  
  ```
  2022-08-26T15:48:11Z Adding fixed seeds as -dnsseed=0, -addnode is not provided and all -seednode(s) attempted
  2022-08-26T15:48:16Z New outbound peer connected: version: 70016, blocks=2344103, peer=0 (block-relay-only)
  2022-08-26T15:48:23Z UpdateTip: new best=0000000000000031a474cee013b177d26526fba93730c2d522fa75f7d25ebea7 height=2344100 version=0x20600000 log2_work=74.890090 tx=63514515 date='2022-08-26T15:07:59Z' progress=0.999997 cache=0.0MiB(79txo)
  2022-08-26T15:48:23Z UpdateTip: new best=000000000000000d3707a388f7e737adf8360b30d67e79297338b50df37defce height=2344101 version=0x20004000 log2_work=74.890104 tx=63514570 date='2022-08-26T15:15:03Z' progress=0.999998 cache=0.0MiB(230txo)
  2022-08-26T15:48:23Z UpdateTip: new best=0000000000000036f5cfdfdcb3f15bd9495dddddf242a63a8696cad6abd37307 height=2344102 version=0x20c00000 log2_work=74.890117 tx=63514666 date='2022-08-26T15:23:46Z' progress=0.999998 cache=0.1MiB(408txo)
  2022-08-26T15:48:23Z UpdateTip: new best=00000000000000249a4fd1b01db85b52b1f92751a5828a23245eb586b9dd7727 height=2344103 version=0x20400000 log2_work=74.890130 tx=63514797 date='2022-08-26T15:42:53Z' progress=1.000000 cache=0.1MiB(699txo)
  2022-08-26T15:48:30Z New outbound peer connected: version: 70016, blocks=2344103, peer=1 (block-relay-only)
  2022-08-26T15:48:32Z New outbound peer connected: version: 70016, blocks=2344103, peer=2 (outbound-full-relay)
  2022-08-26T15:48:37Z New outbound peer connected: version: 70016, blocks=2344103, peer=3 (outbound-full-relay)
  2022-08-26T15:48:46Z New outbound peer connected: version: 70016, blocks=2344103, peer=4 (outbound-full-relay)
  2022-08-26T15:48:52Z New outbound peer connected: version: 70016, blocks=2344103, peer=5 (outbound-full-relay)
  2022-08-26T15:48:59Z New outbound peer connected: version: 70016, blocks=2344103, peer=6 (outbound-full-relay)
  2022-08-26T15:49:12Z Socks5() connect to mrhiniicugfo7mgrwv3wtolk3tptlcw2uq7ih6sq43fa4k4zbilut3yd.onion:18333 failed: host unreachable
  2022-08-26T15:49:20Z New outbound peer connected: version: 70016, blocks=2344103, peer=8 (outbound-full-relay)
  2022-08-26T15:49:27Z New outbound peer connected: version: 70016, blocks=2344103, peer=9 (outbound-full-relay)
  2022-08-26T15:49:41Z New outbound peer connected: version: 70016, blocks=2344103, peer=10 (outbound-full-relay)
  2022-08-26T15:57:03Z Socks5() connect to 3feanxuw2gpypwjkno4gqgddjkhl7gwrg2efgybhriwdgjbedxfm66qd.onion:18333 failed: host unreachable
  2022-08-26T15:57:19Z Socks5() connect to ihitruxewkiwcy7ek27ghbeus3tgbu5xp5e6qxwdkhwqydtgktegcdad.onion:8333 failed: host unreachable
  2022-08-26T15:57:45Z New outbound peer connected: version: 70016, blocks=2344103, peer=11 (outbound-full-relay)
  2022-08-26T16:03:07Z UpdateTip: new best=00000000000008cb62c573b29d70dfdce738e2bff67516e8d3b6b07616cca486 height=2344104 version=0x20000004 log2_work=74.890130 tx=63514909 date='2022-08-26T16:03:00Z' progress=1.000000 cache=0.1MiB(887txo)
  2022-08-26T16:04:06Z Socks5() connect to eis45gyxcexsz76bnnm5flc4jmmwivimdjwfjhahdjybeayrmw5tlgad.onion:18333 failed: host unreachable
  2022-08-26T16:06:43Z New outbound peer connected: version: 70016, blocks=2344104, peer=16 (outbound-full-relay)
  2022-08-26T16:08:28Z Socks5() connect to mrhiniicugfo7mgrwv3wtolk3tptlcw2uq7ih6sq43fa4k4zbilut3yd.onion:18333 failed: host unreachable
  2022-08-26T16:12:59Z New outbound peer connected: version: 70016, blocks=2344104, peer=19 (block-relay-only)
  2022-08-26T16:15:52Z Socks5() connect to jp7l6fey6q36jxgmhqyzjh3na3urkzri4rlhruu5p4favbcpp4zepjyd.onion:18333 failed: host unreachable
  2022-08-26T16:22:41Z Socks5() connect to plous4m7epczek6psr2mzzalw56e6v7kqgkl6ywcgx4llxmsnee2wkad.onion:18333 failed: host unreachable
  2022-08-26T16:23:07Z UpdateTip: new best=00000000e34e495be4cf2c8b0b4508499414767dbc6ab1d54a13d2ece5e027a2 height=2344105 version=0x20000000 log2_work=74.890130 tx=63515061 date='2022-08-26T16:23:05Z' progress=1.000000 cache=0.2MiB(1116txo)
  2022-08-26T16:25:12Z Socks5() connect to lnnqh7gm4uczdfx7fnkml7fpfp7slb62rimjpd2urcctsnazfy5bq2ad.onion:18333 failed: host unreachable
  2022-08-26T16:25:48Z UpdateTip: new best=0000000000000017d6d899f30ae2edba0d79baad95720a645b6d5e78ec1ab313 height=2344106 version=0x20000000 log2_work=74.890143 tx=63515094 date='2022-08-26T16:25:58Z' progress=1.000000 cache=0.2MiB(1134txo)
  2022-08-26T16:26:42Z New outbound peer connected: version: 70016, blocks=2344106, peer=21 (block-relay-only)
  2022-08-26T16:27:46Z New outbound peer connected: version: 70016, blocks=2344106, peer=22 (block-relay-only)
  ```
</details>

***Question***: I see my node trying to connect to an address that was removed from the list of seeds in this PR: 
`Socks5() connect to mrhiniicugfo7mgrwv3wtolk3tptlcw2uq7ih6sq43fa4k4zbilut3yd.onion:18333 failed: host unreachable`.
Why would it do that? Was that address reintroduced by another peer?

4. Things appear good with the watch command you provided, 

```
Bitcoin Core client v23.99.0-4391364a5a8e testnet - server 70016/Satoshi:23.99.0/

<->   type   net  mping   ping send recv  txn  blk  hb addrp addrl  age id address                                                              version
out   full onion    274    323   36   33    1    8  .   1006         89  9 qe2jbe447he6panfvpyqhyntf7346gmuf55bxrmdzggmgwyjsyknhxyd.onion:18333 70016/Satoshi:23.0.0/
out  block onion    400   1007   28   28    *    8  .      .         90  1 vctlwaqgmu53eutz2hewuakcipfgtyljsd7czut4dd62xr3rp6fqezad.onion:18333 70016/Satoshi:23.99.0/
out  block onion    472    508   42   42    *   55  .      .         90  0 dqhhlssfwmh3g6zhwxpcfbw64xz5rfikcglinbhoxv5ajv4qzicjyeid.onion:18333 70016/Satoshi:22.0.0/
out   full onion    486   1407   34   49    1            114         89 10 ew7havgwsrjdu7xyiloy2tvtdg5cqjj4x62ojtem2nt2w7l4jadq7bad.onion:18333 70016/Satoshi:22.0.0/
out   full onion    497   1019   21   20    1           1003         90  3 gy6nih4pmp5esyvvnhlj6qvk7zkbjuoswkxffyiip3dbkvsfxwz5zcqd.onion:18333 70016/Satoshi:0.21.0/
out   full onion    542    725   36   35    0           1002         89  8 otmfnhc6wrrbf2tpdy6zkisqc3r3urnsuowsnmatoto6yixaocnkseid.onion:18333 70016/Satoshi:23.99.0/
out   full onion    594   1025   26   26    0            190         90  2 zcep44k7unwjm2wxty4ijh2e4fv5zgbrvwlctzyaqnrqhltjfzrtodad.onion:18333 70016/Satoshi:22.0.0/
out   full onion    674   1642   34   49    4            165         90  6 2lsncqdflwk272dhydrxf7ikfy23ppnmm54dnynyxiym6lqf3wowrmqd.onion:18333 70016/Satoshi:23.0.0/
out   full onion    843   1127   20   18    1           1005         90  5 aesy6tfufadkut6flu2bsqgnw2422ur2ynjalguxlzuzuktg3zehttqd.onion:18333 70016/Satoshi:0.21.0/
out   full onion   1074   1074   14   14   29            196         72 16 jjfuyj7krgzkmpxvn3b2j2hwlzkmze3ezy3ifwk7dnswwawgmzqhjrqd.onion:18333 70016/Satoshi:22.0.0/
                     ms     ms  sec  sec  min  min                  min

        onion   total   block
in          0       0
out        10      10       2
total      10      10

Local addresses: n/a

```

though I only see two values for the `blk` column, which corresponds to `bitcoin-cli.NetInfoRequestHandler.peer.last_blck` which I believe comes from this line in `net.cpp.getPeerInfo()`:

```
        obj.pushKV("last_block", count_seconds(stats.m_last_block_time));
```
The point I'm trying to make is I believe this column accounts for the number of minutes since the last block so I think I've only gotten blocks from three peers? Is that normal? It seems low but this is testnet over tor so I don't expect things to be as active or robust as non-tor testnet or mainnet.

7. Took my list of nodes (`bitcoin-cli -rpcwait -testnet getnodeaddressess`...) and compared it with the list in this PR (used this command to see there are zero lines unique to nodes_test.txt: `comm -2 -3 nodes_test.txt nodes_test_stacielocal.txt`). All the seeds in this PR are in my list, and the only difference is my list contains addresses that this PR doesn't have (which is to be expected).

8. Ran the script in the PR description to see if the addresses were connectable. I got a handful that I couldn't connect to after running the script twice. Not sure if any changes are warranted though since the uptime percentage for these nodes is not known.

```
3avu2v7xovnd366dnizqrebr7dmulib2togxkfmfzvnr4ympjdq4a7ad.onion:18333
3zxyxhnh75untdyd47xdricmfxhe642n3u5rdaljgpydcrvtwexkieyd.onion:18333
cajjeo4q2pxygu5jlfqvxgsxaemvplkudztn4x2atxmrbb4ylcze7wad.onion:18333
dmbuz5xohpuiojomtpmapsyp4lw3atznrh6wahdvhvlmwgoxdnz2h3id.onion:18333
i5oftdk6lxq4j65vxfstiqqaqw4znqlcci56xnyodbeusay3b5emxgid.onion:18333
y2cpty5imvnbnkkpjfsxfp6l72jvosskz2vecouixnfq227p2ivpdwyd.onion:18333
```

jonatack commented:
> Woo hoo! Thanks for the updated list and test instructions. Can you explain service bit part? I see it in `seeds_main.txt` (and `makeseeds.py` reads the service bit), but how did you get it for the tor nodes? Is it from the "services" field in the `getnodeaddresses` rpc?

Yes, it is the services/service_names fields in RPCs like getnodeaddresses and getpeerinfo and the GUI peers detail area. 

Bit 1 (odd numbers) corresponds to `NODE_NETWORK` in `enum ServiceFlags` in `src/protocol.h`:

```cpp
    // NODE_NETWORK means that the node is capable of serving the complete block chain. It is currently
    // set by all Bitcoin Core non pruned nodes, and is unset by SPV clients or other light clients.
    NODE_NETWORK = (1 << 0),
```

>  1. Regenerated `chainparamsseeds.h`. No differences between my version and the one in this PR.

Perfect :)

>  2. Re-compiled (by running `autogen.sh`, `configure`, and `make`. I forgot to do this the first time I tried to retest and saw a bunch of failed connections to the unreachable seeds that this PR removes) and brought up a new testnet node with the command you provided (also deleted `testnet3/peers.dat` before launching for good measure). 15 min into startup the logs look good. After a little while I see some failed connections, but a spot check revealed none of these peers were in `nodes_test.txt`.

Yes!

> _**Question**_: I see my node trying to connect to an address that was removed from the list of seeds in this PR: `Socks5() connect to mrhiniicugfo7mgrwv3wtolk3tptlcw2uq7ih6sq43fa4k4zbilut3yd.onion:18333 failed: host unreachable`. Why would it do that? Was that address reintroduced by another peer?

I think so.

> 4. Things appear good with the watch command you provided,

:ok_hand:  `watch -t ./src/bitcoin-cli -testnet -netinfo 4`

> though I only see three values for the `blk` column, which corresponds to `bitcoin-cli.NetInfoRequestHandler.peer.last_blck` which I believe comes from this line in `net.cpp.getPeerInfo()`:
> 
> ```
>         obj.pushKV("last_block", count_seconds(stats.m_last_block_time));
> ```

Yes, the `blk` column is the time in minutes since the last novel block passing initial validity checks was received from the peer. See also `bitcoin-cli -netinfo help`.
 
> The point I'm trying to make is I believe this column accounts for the number of minutes since the last block so I think I've only gotten blocks from three peers? Is that normal? It seems low but this is testnet over tor so I don't expect things to be as active or robust as non-tor testnet or mainnet.

Yes, it's normal (depending for example on how long bitcoind has been running and if the same peer sent you multiple novel blocks; "novel" meaning a block that your node had not seen yet).

> 7. Took my list of nodes (`bitcoin-cli -rpcwait -testnet getnodeaddressess`...) and compared it with the list in this PR (used this command to see there are zero lines unique to nodes_test.txt: `comm -2 -3 nodes_test.txt nodes_test_stacielocal.txt`). All the seeds in this PR are in my list, and the only difference is my list contains addresses that this PR doesn't have (which is to be expected).

Awesome.

>     8. Ran the script in the PR description to see if the addresses were connectable. I got a handful that I couldn't connect to after running the script twice. 

Some probably go offline and come back online, but two days after your report I am seeing the same unreachable ones as you, so removing them :)

satsie commented:
ACK 2ef33e936eaf1058086169b5833f196ff624bf89

I checked out your latest commit and repeated all the test steps from my previous comment. Everything was just as smooth as the last time I tested (my node had 8 outbound peers within a minute of the `Adding fixed seeds` log message, no diffs in  `chainparamsseeds.h`, addrman contains the seeds, netinfo RPC shows that blocks and transactions are coming through)

Thanks for the detailed reply and for answering my questions! It was great to see where those service flags are defined in the code.
